bindgen/templates/CallbackInterfaceImpl.cs

```diff
@@ -128,16 +133,17 @@ static void UniffiFree(ulong @handle) {
     {%- endfor %}
     static _UniFFILib.UniffiCallbackInterfaceFree _callback_interface_free = new _UniFFILib.UniffiCallbackInterfaceFree(UniffiFree);
 
-    public static _UniFFILib.{{ vtable|ffi_type_name }} _vtable = new _UniFFILib.{{ vtable|ffi_type_name }} {
-        {%- for (ffi_callback, meth) in vtable_methods.iter() %}
-        {%- let fn_type = format!("_UniFFILib.{}Method", callback_impl_name) %}
-        {{ meth.name()|var_name() }} = Marshal.GetFunctionPointerForDelegate(_m{{ loop.index0 }}),
-        {%- endfor %}
-        @uniffiFree = Marshal.GetFunctionPointerForDelegate(_callback_interface_free)
-    };
-
     public static void Register() {
-        _UniFFILib.{{ ffi_init_callback.name() }}(ref {{ callback_impl_name }}._vtable);
+        _UniFFILib.{{ vtable|ffi_type_name }} _vtable = new _UniFFILib.{{ vtable|ffi_type_name }} {
+            {%- for (ffi_callback, meth) in vtable_methods.iter() %}
+            {%- let fn_type = format!("_UniFFILib.{}Method", callback_impl_name) %}
+            {{ meth.name()|var_name() }} = Marshal.GetFunctionPointerForDelegate(_m{{ loop.index0 }}),
+            {%- endfor %}
+            @uniffiFree = Marshal.GetFunctionPointerForDelegate(_callback_interface_free)
+        };
+
+        // Pin vtable to ensure GC does not move the vtable across the heap
+        _UniFFILib.{{ ffi_init_callback.name() }}(GCHandle.Alloc(_vtable, GCHandleType.Pinned).AddrOfPinnedObject());
```

Comment:
Im not fully familiar on how the GCHandle.Alloc works, will the memory still remain valid even after the handle goes out of scope here?